state-chain/pallets/cf-elections/src/electoral_systems/liveness.rs

```diff
@@ -32,6 +32,24 @@ pub trait OnCheckComplete<ValidatorId> {
 	fn on_check_complete(validator_ids: BTreeSet<ValidatorId>);
 }
 
+#[macro_export]
+macro_rules! create_on_check_complete_hook {
```

Comment:
This looks like it could just be a generic struct. Does something like this work?

```rust
struct ReportFailedLivenessCheck<C>;

impl<C: Get<ForeignChain>> OnCheckComplete<<Runtime as Chainflip>::ValidatorId> for ReportFailedLivenessCheck<C> {
	fn on_check_complete(validator_ids: BTreeSet<<Runtime as Chainflip>::ValidatorId>) {
		<Reputation as OffenceReporter>::report_many(
			Offence::FailedLivenessCheck(C::get()),
			validator_ids,
		);
	}
}

// then in bitcoin_elections.rs:
pub type ChainLiveness = Liveness<
	BlockNumber,
	Hash,
	cf_primitives::BlockNumber,
	ReportFailedLivenessCheck<Bitcoin>,
	<Runtime as Chainflip>::ValidatorId,
>;
```